extensions/qute-resteasy/deployment/src/main/java/io/quarkus/qute/resteasy/deployment/QuteResteasyProcessor.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.qute.resteasy.deployment;
+
+import org.jboss.jandex.DotName;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveHierarchyIgnoreWarningBuildItem;
+import io.quarkus.qute.TemplateInstance;
+import io.quarkus.qute.resteasy.TemplateResponseFilter;
+import io.quarkus.resteasy.common.spi.ResteasyJaxrsProviderBuildItem;
+
+public class QuteResteasyProcessor {
+
+    @BuildStep
+    ResteasyJaxrsProviderBuildItem registerProviders() {
+        return new ResteasyJaxrsProviderBuildItem(TemplateResponseFilter.class.getName());
+    }
+
+    @BuildStep
+    ReflectiveHierarchyIgnoreWarningBuildItem ignoreReflectiveWarning() {
+        return new ReflectiveHierarchyIgnoreWarningBuildItem(DotName.createSimple(TemplateInstance.class.getName()));
```

Comment:
I think it's more that we completely want to ignore it for RESTEasy? Because if for some reason the `TemplateInstance` class is in the index, you really don't want to include it in reflection.

In this case, for now, the type should be added there instead: https://github.com/quarkusio/quarkus/blob/master/extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java#L69 .

We probably should have a better mechanism based on build items but that will do for now.

Reply:
1) well, it shouldn't be in the index, 2) if it is, it's probably not a big problem, or?, 3) I'd like to avoid `ResteasyDotNames.TYPES_IGNORED_FOR_REFLECTION` if possible.